### ISSUE TITLE
Scraping a website that requires you to scroll down?

### ISSUE THREAD

**Issue by Aracki** [gocolly/colly#261]
Is there a way to force the `colly.Colector` to scroll down the page in order to get additional data?

**Comment by zvodd** [comment]
TL:DR - No, read some tutorials or books on webscraping and/or check out http://phantomjs.org / http://seleniumhq.org for automating a complete web browser.

Colly interacts with websites by way of sending HTTP requests for specific resources. Websites that scroll are usually(always?) implemented through JavaScript detecting the scroll interaction and making additional HTTP requests (aka AJAX, XHR) and finally translating received responses back into the page - by manipulating the original HTML document with the JavaScript DOM API  inside the clients browser.

Colly doesn't "resolved"/render the HTML DOM to the same degree as a web browser, in a few ways, but mainly it doesn't interpret and run JavaScript. So there is no way for scroll events to happen or acted upon.

To use Colly to scrape these types of lazy loaded resources, you need to manually identify the pattern of those HTTP resources - the URLs and how they are embedded in / called from the initial page.

**Comment by Aracki** [comment]
Thanks for a very profound answer. 👍

**Comment by roopeshsn** [comment]
Whether this case is still valid now? or does Colly have this functionality?
